src/main/java/org/folio/uk/integration/kafka/KafkaMessageListener.java

```diff
@@ -44,24 +42,21 @@ public class KafkaMessageListener {
     topicPattern = "#{folioKafkaProperties.listener['system-user'].topicPattern}")
   public void handleSystemUserEvent(ResourceEvent event) {
     log.info("System user event received: {}", event);
-    var eventType = event.getType();
 
-    if (eventType != ResourceEventType.CREATE) {
-      return;
-    }
-
-    var folioHeaders = prepareFolioHeaders(event);
-    try (var ignored = new FolioExecutionContextSetter(metadata, folioHeaders)) {
-      var sysUserEvent = objectMapper.convertValue(event.getNewValue(), SystemUserEvent.class);
-      asyncTaskExecutor.execute(getRunnableWithCurrentFolioContext(
-        () -> systemUserService.createOnEvent(sysUserEvent)));
+    switch (event.getType()) {
+      case UPDATE -> executeWithContext(event,
+        () -> systemUserService.updateOnEvent(objectMapper.convertValue(event.getNewValue(), SystemUserEvent.class)));
+      case CREATE -> executeWithContext(event,
+        () -> systemUserService.createOnEvent(objectMapper.convertValue(event.getNewValue(), SystemUserEvent.class)));
+      case DELETE -> executeWithContext(event, systemUserService::delete);
+      default -> log.warn("Received system user event is not handled: {}", event);
     }
   }
 
-  private Map<String, Collection<String>> prepareFolioHeaders(ResourceEvent event) {
-    return Map.of(
-      TENANT, List.of(event.getTenant()),
-      URL, List.of(okapiProperties.getUrl())
-    );
```

Comment:
what is the benefit of removing this method? Because, the code below shows that it does not fit in one line, which makes that part of the code harder to read

Reply:
discussed offline